airflow/jobs/job.py

```diff
@@ -207,7 +207,16 @@ def heartbeat(
                 self.log.debug("[heartbeat]")
         except OperationalError:
             Stats.incr(convert_camel_to_snake(self.__class__.__name__) + "_heartbeat_failure", 1, 1)
-            self.log.exception("%s heartbeat got an exception", self.__class__.__name__)
+            if self.is_alive():
+                self.log.error(
+                    "%s heartbeat failed with error. Scheduler may go into unhealthy state",
```

Comment:
Maybe explain in more detail when (if it more than occasional error).

Reply:
@potiuk Do you want to explain in detail why the scheduler may go to an unhealthy state? If yes, how do I find the list of possible cases when it may go into the unhealthy state?

Reply:
I think, again maybe @BShraman can help - they seem to be engaged and propose improvement in this area and possibly they have some experience and can propose concrete wording and explanation here and list of cases. Also possibly @hterik - the creator of #31810 has some past experiences?

I think what might be best is to add a section in the documentation (in scheduler docs) describing some details about possible heartbeat loss and error situations that might happen here and rather than explaining them in details in this log message, it could be explained there and link to that documentation section could be added as". See more at http://airflow.apache.org/....." in the message.